homeassistant/components/switch_as_x/__init__.py

```diff
@@ -32,6 +33,24 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         )
         return False
 
+    async def async_registry_updated(event: Event) -> None:
+        """Handle entity registry update."""
+        data = event.data
+        if data["action"] == "remove":
+            await hass.config_entries.async_remove(entry.entry_id)
```

Comment:
Wonder if this is the most expected way to handle this for a user or if it becoming unavailable is more expected?

Reply:
I'm not entirely sure either.
What would be the benefit of keeping the config entry around when the tracked entity is removed?

Reply:
Yeah, not sure. I think it's ok like this.